src/inference/include/openvino/runtime/intel_gpu/properties.hpp

```diff
@@ -115,6 +115,14 @@ static constexpr Property<ov::hint::Priority> host_task_priority{"GPU_HOST_TASK_
  * @ingroup ov_runtime_ocl_gpu_prop_cpp_api
  */
 static constexpr Property<int64_t> available_device_mem{"AVAILABLE_DEVICE_MEM_SIZE"};
+
+/**
+ * @brief This key specifies priority for optimization in GPU plugin for dynamic shapes either of memory or performance.
+ * - "memory" : Set memory management policy to optimize memory aggresively, which can affect peformmance drop.
+ * - "performance" : Prioritize performance always, which can increase memory usage.
+ * @ingroup ov_runtime_ocl_gpu_prop_cpp_api
+ */
+static constexpr Property<std::string> llm_opt_mode{"GPU_LLM_OPT_MODE"};
```

Comment:
1. It shouldn't have string type. Some enum is needed.
2. the behavior is not limited by llm, so the name is not fully relevant

Reply:
Maybe
```
static constexpr Property<bool> enable_aggressive_memory_opt;
```

Reply:
Updated.